src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Performance/UseSearchValues.cs

```diff
@@ -0,0 +1,314 @@
+                if (method.Name is "IndexOfAny" or "LastIndexOfAny" &&
+                    method.Parameters.Length == 1)
+                {
+                    methods.Add(method);
+                }
+            }
+
+            // {ReadOnly}Span<T>.{Last}IndexOfAny{Except}(ReadOnlySpan<T>) and ContainsAny{Except}(ReadOnlySpan<T>)
+            if (compilation.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemMemoryExtensions, out var memoryExtensionsType) &&
+                compilation.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemSpan1, out var spanType) &&
+                compilation.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemReadOnlySpan1, out var readOnlySpanType))
+            {
+                foreach (var method in memoryExtensionsType.GetMembers().OfType<IMethodSymbol>())
+                {
+                    if (method.Parameters.Length != 2 || method.TypeParameters.Length != 1)
+                    {
+                        continue;
+                    }
+
+                    if (method.Name is not ("IndexOfAny" or "IndexOfAnyExcept" or "LastIndexOfAny" or "LastIndexOfAnyExcept" or "ContainsAny" or "ContainsAnyExcept"))
+                    {
+                        continue;
+                    }
+
+                    var firstParameterType = method.Parameters[0].Type.OriginalDefinition;
+                    var secondParameterType = method.Parameters[1].Type.OriginalDefinition;
+
+                    if (!SymbolEqualityComparer.Default.Equals(firstParameterType, spanType) &&
+                        !SymbolEqualityComparer.Default.Equals(firstParameterType, readOnlySpanType))
+                    {
+                        continue;
+                    }
+
+                    if (!SymbolEqualityComparer.Default.Equals(secondParameterType, readOnlySpanType))
+                    {
+                        continue;
+                    }
+
+                    // All of these methods are generic for any T, but SearchValues only supports byte/char variants.
+                    methods.Add(method.Construct(compilation.GetSpecialType(SpecialType.System_Byte)));
+                    methods.Add(method.Construct(compilation.GetSpecialType(SpecialType.System_Char)));
+                }
+            }
+
+            return methods;
+        }
+
+        // It's not always worth going through SearchValues if there are very few values used
+        // such that they will already be using dedicated vectorized paths.
+        private const int MinLengthWorthReplacing = 6;
+
+        private bool AreConstantValuesWorthReplacingForStringIndexOfAny(IOperation argument)
+        {
+            if (argument is IArrayCreationOperation arrayCreation)
+            {
+                // text.IndexOfAny(new[] { 'a', 'b', 'c' })
+                return IsConstantByteOrCharSZArrayCreation(arrayCreation, out _);
+            }
+            else if (argument is IFieldReferenceOperation fieldReference)
```

Comment:
We only handle field references in this method, while we handle both field and property references in `AreConstantValuesWorthReplacing`. Any reason we don't support both of them here?

Reply:
With `Span`-based overloads below, the properties we're looking for might look like
```c#
ReadOnlySpan<byte> MyProperty => new byte[] { ... };

text.IndexOfAny(MyProperty);
```
which is a reasonable thing someone might write if they know that allocation will be eliminated by the compiler.

For `string`, the property would instead have to return a `char[]`, which felt like a really odd pattern to use. We could support it though.